### ISSUE TITLE
VQ-Bet unexpected shape error

### ISSUE THREAD

**Issue by dennisushi** [huggingface/lerobot#348]
### System Info

```Shell
8df7e63, Ubuntu 22, Python 3.10
```

### Information

- [ ] One of the scripts in the examples/ folder of LeRobot
- [X] My own task or dataset (give details below)

### Reproduction

TLDR: _Is it essential for VQ-Bet to have same action and state space? Or is this a bug?_

I am using a custom environment (StackCube-v1 from Maniskill3), I made a LeRobot dataset, it trains errorless on ACT (though the results are not what I expected but that is not the issue here). One peculiar thing about this environment is that the state (joint positions) has 9 dimensions while the action has 8 (grasping is True/False, instead of joint values). 

My question relates to this line:
https://github.com/huggingface/lerobot/blob/8df7e63d61af6c064a510426b4a1814fcc15ce7b/lerobot/common/policies/vqbet/modeling_vqbet.py#L291-L292
 
If the state projector takes in the action dimensions, this causes an error after the second training phase starts (i.e. discretization does not fail, but state projection does). 

Looking at the VQ-Bet paper, it seems that it uses actions and observations but no low dimensional state conditioning. Shouldn't the code:
- 1) produce an error if there's such mismatch, if it is not allowed? 
- 2) use the actions in the input directly, if the state is to be ignored?

Here is the error log:
```
  File "/home/dennisushi/repos/3dp/lerobot/lerobot/scripts/train.py" in train_cli
    train(
  File "/home/dennisushi/repos/3dp/lerobot/lerobot/scripts/train.py" in train
    train_info = update_policy(
  File "/home/dennisushi/repos/3dp/lerobot/lerobot/scripts/train.py" in update_policy
    output_dict = policy.forward(batch)
  File "/home/dennisushi/repos/3dp/lerobot/lerobot/common/policies/vqbet/modeling_vqbet.py" in forward
    _, loss_dict = self.vqbet(batch, rollout=False)
  File "/home/dennisushi/micromamba_envs/envs/diffusion_policy_v2/lib/python3.9/site-packages/torch/nn/modules/module.py" in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/dennisushi/repos/3dp/lerobot/lerobot/common/policies/vqbet/modeling_vqbet.py" in forward
    self.state_projector(batch["observation.state"])
  File "/home/dennisushi/micromamba_envs/envs/diffusion_policy_v2/lib/python3.9/site-packages/torch/nn/modules/module.py" in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/dennisushi/micromamba_envs/envs/diffusion_policy_v2/lib/python3.9/site-packages/torch/nn/modules/container.py" in forward
    input = module(input)
  File "/home/dennisushi/micromamba_envs/envs/diffusion_policy_v2/lib/python3.9/site-packages/torch/nn/modules/module.py" in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/dennisushi/micromamba_envs/envs/diffusion_policy_v2/lib/python3.9/site-packages/torch/nn/modules/linear.py" in forward
    return F.linear(input, self.weight, self.bias)
RuntimeError: mat1 and mat2 shapes cannot be multiplied (320x9 and 8x512)
```

### Expected behavior

Capability of having different state and action spaces, or raised error.

**Comment by alexander-soare** [comment]
@dennisushi thanks for reporting an my apologies for the delayed response.

Pretty sure that we should just change that key on like 292 to "state" instead of action. Would that solve the problem? cc @jayLEE0301

**Comment by jayLEE0301** [comment]
@dennisushi  Thank you for reporting the error.
@alexander-soare  I agree. It could be changed as
```
MLP(config.input_shapes["observation.state"][0], hidden_channels=[self.config.gpt_input_dim] 
```

**Comment by alexander-soare** [comment]
@jayLEE0301 thanks for confirming. @dennisushi if you'd like to contribute, you're welcome to make this change. I'll do it later this week otherwise.

**Comment by alexander-soare** [comment]
Solved by #365
